packages/core/src/components/Table/TableVirtualizedBody/TableVirtualizedBody.tsx

```diff
@@ -19,66 +19,72 @@ export interface ITableVirtualizedBodyProps extends VibeComponentProps {
   onScroll?: (horizontalScrollDirection: ScrollDirection, scrollTop: number, scrollUpdateWasRequested: boolean) => void;
 }
 
-const TableVirtualizedBody: FC<ITableVirtualizedBodyProps> = ({
-  items,
-  rowRenderer,
-  onScroll,
-  id,
-  className,
-  "data-testid": dataTestId
-}) => {
-  const { size, rowWidth } = useTable();
+const TableVirtualizedBody = forwardRef(
+  (
+    { items, rowRenderer, onScroll, id, className, "data-testid": dataTestId }: ITableVirtualizedBodyProps,
+    ref: React.ForwardedRef<HTMLDivElement>
+  ) => {
+    const { size, virtualizedListRef, onVirtualizedListScroll, markTableAsVirtualized } = useTable();
 
-  const itemRenderer = useCallback<ComponentType<ListChildComponentProps<TableVirtualizedRow>>>(
-    ({ index, style }) => {
-      const currentItem = items[index];
-      const element = rowRenderer(currentItem);
-      return React.cloneElement(element, { style, key: index });
-    },
-    [items, rowRenderer]
-  );
+    useEffect(() => {
+      markTableAsVirtualized();
```

Comment:
Not sure I got this, can you elaborate?

Reply:
wanted to use context's power to prevent user from passing another prop to table
so that's the way I went with, to let other components within table they should behave in a virtualized variant